### PR TITLE
feat(components): allow Actions to contain ActionSplitDropdown

### PR DESCRIPTION
**Pull request by ncomont**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Actions cannot contains ActionSplitDropdown.

**What is the new behavior?**
Actions can contains ActionSplitDropdown by passing 'splitDropdown' as displayMode.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://343.talend.surge.sh)

File: packages/components/src/Actions/Actions.component.js [hunk 1]
```diff
@@ -49,32 +53,27 @@ const actions: [
...
+					key: index,
+					...globalParams,
+					...rest,
+				};
+				const ActionType = {
+					[TYPE_DROPDOWN]: ActionDropdown,
+					[TYPE_SPLIT_DROPDOWN]: ActionSplitDropdown,
+				}[displayMode] || Action;
+
+				return <ActionType {...params} />;
```
Comment (talend-glorieux) [inline packages/components/src/Actions/Actions.component.js]: The ActionType looks smaller indeed but it makes the code harder to understand and does an extra allocation for every resolution of the ActionType. 
Id rather see a function returning the ActionType Component with a proper switch/case.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://343.talend.surge.sh)

File: packages/components/stories/Actions.js [hunk 2]
```diff
@@ -36,6 +36,22 @@ const actions = [
 			},
 		],
 	},
+	{
```
Comment (charlesnguyen) [inline packages/components/stories/Actions.js]: splitDropdown is not correctly rendered in "vertical" display cases

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://343.talend.surge.sh)

File: packages/components/src/Actions/Actions.component.js [hunk 3]
```diff
@@ -15,6 +19,17 @@ function getButtonGroupProps(props) {
 	return buttonGroupProps;
 }
 
+function getActionType(displayMode) {
```
Comment (jmfrancois) [inline packages/components/src/Actions/Actions.component.js]: rename: getActionComponent

File: packages/components/src/Actions/Actions.component.js [hunk 4]
```diff
@@ -43,38 +58,46 @@ const actions: [
...
-							link={props.link}
-							tooltipPlacement={props.tooltipPlacement}
-							{...rest}
-						/>
-					);
-				}
+				const ActionType = getActionType(displayMode);
+				const params = {
+					key: index,
+					...globalParams,
```
Comment (jmfrancois) [inline packages/components/src/Actions/Actions.component.js]: question: why did you need globalParams please ?

File: packages/components/src/Actions/Actions.component.js [hunk 5]
```diff
@@ -85,9 +108,13 @@ Actions.propTypes = {
 		PropTypes.oneOfType([
 			PropTypes.shape(Action.propTypes),
 			PropTypes.shape({
-				displayMode: PropTypes.oneOf([TYPE_DROPDOWN]),
+				displayMode: TYPE_DROPDOWN,
```
Comment (jmfrancois) [inline packages/components/src/Actions/Actions.component.js]: error: shape is waiting key / proptypes.
proptypes is should be PropTypes.string or oneOf as before.

Comment (ncomont) [inline packages/components/src/Actions/Actions.component.js]: Oh, sorry, it was usefull before my last fix (https://github.com/Talend/ui/pull/343#pullrequestreview-30278497). I'll delete it.

File: packages/components/src/Actions/Actions.component.js (hunk 4 above)
Comment (ncomont) [inline packages/components/src/Actions/Actions.component.js]: Oh, sorry, it was usefull before my last fix (#343 (review)). I'll delete it.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://343.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://343.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://343.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://343.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://343.talend.surge.sh)
